pennylane/tape/qscript.py

```diff
@@ -226,6 +228,12 @@ def __len__(self):
     # QSCRIPT properties
     # ========================================================
 
+    @property
+    def measured_wires(self) -> List[int]:
+        """Returns the measured wires"""
+        self._update_measured_wires()
```

Comment:
as mentioned below, the question is to decide when we need to re-evaluate this. I'm fine with re-evaluating whenever someone requests it, but at that point there is no need for the private `_measured_wires` property - just return whatever you evaluate!

Reply:
This was done based on [albi3ro](https://github.com/albi3ro)'s feedback above. He pointed to `samples_computational_basis` as an example for the desired behaviour.

Reply:
sorry, I think we've got our wires crossed 😆 the `@property` attribute is perfect, I'm more saying that you don't need to save the value to `_measured_wires`. Instead, you can do:
```python
@property
def measured_wires(self) -> Wires:
    """Returns the measured wires"""
    wires = []
    for m in self.measurements:
        if m.obs is not None:
            wires.append(m.obs.wires)
    return Wires.all_wires([*wires, self._measured_wires])
```
and delete the `_update_measured_wires` private method. See the `observables` property for something similar.

Reply:
Now that `_update_measured_wires` isn't used in `_update`, this makes sense.

Reply:
I've made this change in a recent commit.